assets/js/hooks/useChecks.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * `useChecks` hook.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+const { useSelect } = Data;
+
+/**
+ * Runs a series of asynchronous checks returning the first encountered error.
+ * All checks should be functions that throw their respective errors.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array} checks Array of functions to run.
+ * @return {Object} An object containing complete and error properties.
+ */
+export function useChecks( checks ) {
+	const isSiteKitConnected = useSelect( ( select ) => select( CORE_SITE ).isConnected() );
```

Comment:
Thanks for catching this – I think this detail was missed in the AC/IB but I can see that the current `CompatibilityChecks` skips the checks if Site Kit is already connected.

However, this hook is intended to be completely independent of the check logic and is no longer specifically related to setup – it just happens to be the only use for it right now.

In this case, the `isSiteKitConnected` select and state here should move back to the `CompatibilityChecks` component, and if `true` it would provide an empty array of `checks` to `useChecks` rather than the normal checks. Then internally here in the hook, we can default `complete` to `checks.length === 0`, and skip calling `runChecks` all together if `complete` (which would only happen if there are none). Perhaps this logic could live in that `createCompatibilityChecks` function as it will already have access to the registry.